### PR TITLE
fix(synthetic_test): treat empty ID as not-found in Read

### PR DESCRIPTION
**Pull request by nir-sagi**
# User description
GetSyntheticTest("") builds a request to GET /api/synthetics/v1/rules/{id} with an empty id, which redirects (301) to the collection route and returns 200 with the full list — never a 404. So Read mapped an ID-less list response into state and reported the resource as existing.

This breaks upjet/Crossplane, whose pre-create existence check reads with an empty id: it sees resourceExists=true with no id, fails to derive the external-name ("cannot find id in tfstate"), and re-creates on every reconcile (leaking duplicate synthetic tests).

Guard both layers: Read removes the resource from state when the id is empty, and GetSyntheticTest returns ErrNotFound for an empty id so no other caller mistakes the list redirect for an existing resource. Add a unit test.

## Checklist

- [ ] I have written acceptance tests for my changes
- [ ] I have run `make generate` to update generated docs
- [ ] I have added examples demonstrating the new functionality
- [ ] I have updated the README.md with details of changes
- [ ] I have updated the CHANGELOG.md file

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
Ensure the provider’s synthetic test client and resource Read treat empty IDs as not-found so callers don’t mistake the list redirect for an existing synthetic test. Document the fix and cover the empty-id path with a unit test.
<table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/groundcover-com/terraform-provider-groundcover/113?tool=ast&topic=Empty+ID+flow>Empty ID flow</a>
        </td><td>Guard <code>GetSyntheticTest</code> and Read to return not-found when the synthetic test ID is blank so an empty request doesn’t impersonate an existing resource.<details><summary>Modified files (2)</summary><ul><li>internal/provider/client_syntheticstest.go</li>
<li>internal/provider/resource_syntheticstest.go</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>nir.sagi@groundcover.com</td><td>fix(synthetic_test): t...</td><td>July 07, 2026</td></tr>
<tr><td>avital@groundcover.com</td><td>fix synthetics asserti...</td><td>June 08, 2026</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/groundcover-com/terraform-provider-groundcover/113?tool=ast&topic=Tests+%26+changelog>Tests &amp; changelog</a>
        </td><td>Document the behavior change and add a regression test that covers the empty-ID scenario.<details><summary>Modified files (2)</summary><ul><li>CHANGELOG.md</li>
<li>internal/provider/resource_syntheticstest_test.go</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>nir.sagi@groundcover.com</td><td>fix changelog</td><td>July 07, 2026</td></tr>
<tr><td>amir9339@gmail.com</td><td>feat: deprecate ground...</td><td>June 25, 2026</td></tr></table></details></td></tr></table>
<sub><a href="https://baz.co/changes/groundcover-com/terraform-provider-groundcover/113?tool=ast">Review this PR on Baz</a> | <a href="https://baz.co/agents/baz">Customize your next review</a></sub>

## Summary by CodeRabbit

* **Bug Fixes**
  * Empty synthetic test IDs are now treated as not-found, avoiding “collection/list” style responses for individual-resource lookups.
  * Synthetic test read/refresh now removes the resource from state when an empty ID is provided, preventing incorrect existence checks.
* **Tests**
  * Added unit coverage to verify empty-ID requests return the expected not-found result without triggering SDK client calls.
* **Documentation**
  * Updated the changelog for version 1.17.1 describing the empty-ID not-found behavior fix.

### CODE REVIEW

**Review by AvitalTamir — APPROVED** [review]

File: internal/provider/client_syntheticstest.go [hunk 1]
```diff
@@ -35,6 +35,13 @@ func (c *SdkClientWrapper) GetSyntheticTest(ctx context.Context, id string) (*mo
 	logFields := map[string]any{"req": "get_synthetic_test", "id": id}
 	tflog.Debug(ctx, "Executing SDK Call: Get Synthetic Test", logFields)
 
+	// An empty ID would otherwise be sent to GET /rules/{id}, which redirects to the
+	// collection route and returns 200 with the full list — never a 404. Treat it as
+	// not-found so callers don't mistake a list response for an existing resource.
+	if id == "" {
+		return nil, ErrNotFound
+	}
```
[the remark this answers is not in the document]
Comment (AvitalTamir) [reply internal/provider/client_syntheticstest.go:43]: @nir-sagi right - I missed this

**Review by AvitalTamir — CHANGES_REQUESTED** [review]
Add to changeling

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 1.17.1
+
+* Fixed `groundcover_synthetic_test` reads with an empty ID: `GET /api/synthetics/v1/rules/{id}` with an empty id redirects (`301`) to the collection route and returns `200` with the full list — never a `404` — so `Read` mapped an ID-less list response into state and reported the resource as existing. `Read` now removes the resource from state when the ID is empty, and the SDK client returns not-found for an empty ID. This unblocks the groundcover Crossplane provider, whose pre-create existence check reads with an empty id and was otherwise stuck failing to derive the external-name (`cannot find id in tfstate`) and re-creating on every reconcile. No change to normal Terraform create/read/update/delete behaviour
```
Comment (AvitalTamir) [inline CHANGELOG.md]: @nir-sagi I'm not crazy about us adding these AI slop changelog entires. We had a few of these recently and I'm trying to avoid them. Can you write this in a concise sentence without getting into too many technicalities?

**Review by AvitalTamir — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (AvitalTamir) [reply CHANGELOG.md]: something like "Fix synthetics reads with Empty IDs returning 200" would be nicer

Comment (AvitalTamir) [reply CHANGELOG.md]: This ends up in the Terraform docs and provider website and we wanna keep this human readable

**Review by AvitalTamir — APPROVED** [review]
